package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -455,31 +474,35 @@ def guess_donors(self, select='all', max_charge=-0.5):
             hydrogens_sel = self.hydrogens_sel
         hydrogens_ag = self.u.select_atoms(hydrogens_sel)
 
-        ag = hydrogens_ag.residues.atoms.select_atoms(
-            "({donors_sel}) and around {d_h_cutoff} {hydrogens_sel}".format(
-                donors_sel=select,
-                d_h_cutoff=self.d_h_cutoff,
-                hydrogens_sel=hydrogens_sel
+        # We're using u._topology.bonds rather than u.bonds as it is a million times faster to access.
+        # This is because u.bonds also calculates properties of each bond (e.g bond length).
+        # See https://github.com/MDAnalysis/mdanalysis/issues/2396#issuecomment-596251787
+        if (hasattr(self.u._topology, 'bonds') and len(self.u._topology.bonds.values) != 0):
+            donors_ag = find_hydrogen_donors(hydrogens_ag)
```

Comment:
I think I'm missing something obvious here, sorry. Doesn't this ignore `select`?

Reply:
Good point!